### ISSUE TITLE
NetworkInterface.GetPhysicalAddress().GetAddressBytes() seems problematic with LLVM enabled on Android x86_64

### ISSUE THREAD

**Issue by pjcollins** [dotnet/runtime#75155]
### Description

We have a test in xamarin-android that [compares network interface information](https://github.com/xamarin/xamarin-android/blob/5ebac332adc967b944d5f893ebc73c266ef62624/tests/Mono.Android-Tests/System.Net/NetworkInterfaces.cs#L105) obtained from mono vs java that is producing inconsistent results. When this test runs on an x86_64 emulator without AOT+LLVM enabled it passes, but when LLVM is enabled it fails. I have trimmed this down to a smaller repro project that I have attached. The problematic code appears to be as follows:

```csharp
foreach (var inf in NetworkInterface.GetAllNetworkInterfaces ()) {
    byte[] bytes = inf.GetPhysicalAddress ().GetAddressBytes ();
    if (bytes.All (x => x == 0)) {
      // Condition is true for interfaces with physical addresses when LLVM is enabled, and false when LLVM is disabled
    }
}
```

### Reproduction Steps

1. Build xamarin-android/main, or install the workload packs produced by a recent [build from CI](https://devdiv.visualstudio.com/DevDiv/_build?definitionId=11410).

2. Download the [DroidTestIface.zip](https://github.com/dotnet/runtime/files/9499857/DroidTestIface.zip) repro project and run it on an x86_64 emulator:

```
dotnet build DroidTestIface.csproj -v:n -c Release -p:EnableLlvm=true -p:AndroidEnableProfiledAot=false
```

3. After installation open the app on the emulator and tap the button.

### Expected behavior

When building in debug without AOT/LLVM enabled, I see that a couple of network interfaces with physical addresses will return false when checking if their physical address bytes are all zero.

### Actual behavior

When building with AOT/LLVM enabled, I see that the network interfaces that had physical addresses will return true when checking if their physical address bytes are all zero.

### Configuration

This issue is present when running on an an Android x86_64 emulator, using an Android SDK built against the following:

```xml
<Dependency Name="Microsoft.Dotnet.Sdk.Internal" Version="7.0.100-rc.2.22426.5">
  <Uri>https://github.com/dotnet/installer</Uri>
  <Sha>2d1a4de6b2da59647a9538bf9cc4e252847e6231</Sha>
</Dependency>
<Dependency Name="Microsoft.NETCore.App.Ref" Version="7.0.0-rc.1.22422.12" CoherentParentDependency="Microsoft.Dotnet.Sdk.Internal">
  <Uri>https://github.com/dotnet/runtime</Uri>
  <Sha>ef077d0b58ffddcf54fa73bd85dace6b999b8992</Sha>
</Dependency>
```

**Comment by marek-safar** [comment]
I think this is worth backporting either way.

@SamMonoRT could you assign this?

**Comment by SamMonoRT** [comment]
@vargaz can you take a look?

**Comment by vargaz** [comment]
Probably fixed by:
https://github.com/dotnet/runtime/pull/74991
which is not in rc2

**Comment by steveisok** [comment]
> Probably fixed by: #74991 which is not in rc2

@vargaz this should be in RC2 at least according to the date the PR was merged.  Is this good to close?

**Comment by vargaz** [comment]
It is actually in rc2, can somebody check whenever it works now ?
